2011-07-04  Aurelien Jarno  <aurelien@aurel32.net>

	* sysdeps/powerpc/dl-tls.h: Add _PPC_DL_TLS_H inclusion
	protection macro.
	* sysdeps/powerpc/powerpc64/dl-irel.h(Elf64_FuncDesc): include
	<ldsodefs.h> and <dl-machine.h>.
	* sysdeps/powerpc/powerpc64/dl-irel.h(Elf64_FuncDesc): remove.

---
 sysdeps/powerpc/dl-tls.h            |    4 ++++
 sysdeps/powerpc/powerpc64/dl-irel.h |    9 ++-------
 2 files changed, 6 insertions(+), 7 deletions(-)

--- a/sysdeps/powerpc/dl-tls.h
+++ b/sysdeps/powerpc/dl-tls.h
@@ -17,6 +17,8 @@
    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
    02111-1307 USA.  */
 
+#ifndef _PPC_DL_TLS_H
+# define _PPC_DL_TLS_H 1
 
 /* Type used for the representation of TLS information in the TOC.  */
 typedef struct
@@ -47,3 +49,5 @@
 # define GET_ADDR_OFFSET	(ti->ti_offset + TLS_DTV_OFFSET)
 # define __TLS_GET_ADDR(__ti)	(__tls_get_addr (__ti) - TLS_DTV_OFFSET)
 #endif
+
+#endif /* dl-tls.h */
--- a/sysdeps/powerpc/powerpc64/dl-irel.h
+++ b/sysdeps/powerpc/powerpc64/dl-irel.h
@@ -23,16 +23,11 @@
 
 #include <stdio.h>
 #include <unistd.h>
+#include <ldsodefs.h>
+#include <dl-machine.h>
 
 #define ELF_MACHINE_IRELA	1
 
-typedef struct
-{
-  Elf64_Addr fd_func;
-  Elf64_Addr fd_toc;
-  Elf64_Addr fd_aux;
-} Elf64_FuncDesc;
-
 static inline Elf64_Addr
 __attribute ((always_inline))
 elf_ifunc_invoke (Elf64_Addr addr)
